packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -63,6 +63,7 @@ abstract class OverlayRoute<T> extends Route<T> {
   }
 }
 
+/// A route with an entrance transition driven by an [AnimationController].
```

Comment:
There's two transitions that impact a TransitionRoute, each used two different ways.

They are the transition that involves the previous route, and the transition that involves the next route.

The transition that involves the previous route is the transition owned by this TransitionRoute. It's the entrance transition when coming to this route from the previous route, and (in reverse) it's the exit transition when leaving this route and going back to the previous route.

The transition that involves the next route is owned by that next route and passed in when appropriate. It's the exit transition for this route when that other route is being shown on top, and it's the reentrance transition of this route when the next route is going back away and this route is coming back.